Crate fails to compile on Windows due to using `P::display()` on `P: AsRef<Path>`

I'm currently trying to upgrade from 0.11. 0.12 has some other error on Windows. I get this error with both 0.13 and 0.16:

```
   Compiling signatory v0.16.0
error[E0599]: no method named `display` found for type `P` in the current scope
  --> C:\Users\me\.cargo\registry\src\github.com-1ecc6299db9ec823\signatory-0.16.0\src\encoding\encode.rs:90:62
   |
90 |                 Some(&format!("couldn't create {}: {}", path.display(), e)),
   |                                                              ^^^^^^^

error[E0599]: no method named `display` found for type `P` in the current scope
   --> C:\Users\me\.cargo\registry\src\github.com-1ecc6299db9ec823\signatory-0.16.0\src\encoding\pkcs8.rs:101:62
    |
101 |                 Some(&format!("couldn't create {}: {}", path.display(), e)),
    |                                                              ^^^^^^^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0599`.
error: Could not compile `signatory`.
warning: build failed, waiting for other jobs to finish...
error: build failed
PS C:\Users\me\Documents\dev\rust\ire> rustc --version
rustc 1.38.0 (625451e37 2019-09-23)
```

0.12, 0.13, and 0.16 all compile fine on Ubuntu under WSL.

--- a/src/encoding/encode.rs
+++ b/src/encoding/encode.rs
@@ -84,7 +84,8 @@ pub trait Encode: Sized {
         P: AsRef<Path>,
         E: Encoding,
     {
-        let mut file = File::create(path.as_ref()).map_err(|e| {
+        let path = path.as_ref();
+        let mut file = File::create(path).map_err(|e| {
             Error::new(
                 ErrorKind::Io,
                 Some(&format!("couldn't create {}: {}", path.display(), e)),

